File: demo/src/pages/OceanList/OceanList.swift
import SwiftUI

struct OceanList: View {
    struct Ocean: Identifiable, Hashable {
        let name: String
        let id = UUID()
    }

    private var oceans = [
        Ocean(name: "Pacific"),
        Ocean(name: "Atlantic"),
        Ocean(name: "Indian"),
        Ocean(name: "Southern"),
        Ocean(name: "Arctic")
    ]

    @State private var multiSelection = Set<UUID>()

    var body: some View {
          NavigationView {
            List(oceans, selection: $multiSelection) {
                Text($0.name)
            }
            .navigationTitle("Oceans")
            .toolbar { EditButton() }
        }
        Text("\(multiSelection.count) selections")
    }
}

File: demo/src/pages/OceanList/OceanList.ts
//https://developer.apple.com/documentation/swiftui/list/

import {
  Bound,
  EditButton,
  ID,
  Hashable,
  Identifiable,
  List,
  NavigationView,
  Set,
  State,
  swifty,
  Text,
  UUID,
  Viewable,
  CountSet,
  VStack,
  main,
  SelectionType,
  Bindable,
  HasSelection,
  ListConfig,
} from "@tswift/ui";

class OceanClass implements Identifiable, Hashable {
  public id = UUID();
  constructor(public name: string) {}
}

const Ocean = swifty(OceanClass);
@main
export class OceanList extends Viewable {
  oceans: OceanClass[] = [
    Ocean("Pacific"),
    Ocean("Atlantic"),
    Ocean("Indian"),
    Ocean("Southern"),
    Ocean("Arctic"),
  ];

  @State multiSelection = Set<string>();

  body = ({ $multiSelection, oceans, multiSelection }: Bound<this>) => [
    NavigationView(
      List({
        data: oceans,
        selection: $multiSelection,
        content: ($0) => Text($0.name),
      } as ListConfig<OceanClass>)
        .navigationTitle("Oceans")
        .toolbar(EditButton())
    ),
    Text(`${multiSelection.count} selections`),
  ];
}
